src/index.ts

```diff
@@ -770,8 +795,14 @@ export class ValidationService {
     /**
      * Scans the entire document for input elements to be validated.
      */
-    private scanInputs() {
-        let inputs = document.querySelectorAll('[data-val="true"]');
+    private scanInputs(root: HTMLElement) {
+        let inputs = Array.prototype.slice.call(root.querySelectorAll<HTMLElement>('[data-val="true"]')) as HTMLElement[];
```

Comment:
```suggestion
        let inputs = Array.from(root.querySelectorAll<HTMLElement>('[data-val="true"]'));
```

Reply:
Yeah, I tried this but TypeScript was unhappy. Perhaps the `tsconfig.json` is configured for an older set of APIs.

Reply:
Seems like `NodeList` should be `ArrayLike` 🤷‍♂️ 

https://github.com/microsoft/TypeScript/blob/c6bef3f02874bddf6df120aa4f0d130c58478468/lib/lib.es2015.core.d.ts#L72
https://github.com/microsoft/TypeScript/blob/a7548745d78858f94f46f643b2a6edcec12e3a91/src/lib/es5.d.ts#L1503-L1506
https://github.com/microsoft/TypeScript/blob/c6bef3f02874bddf6df120aa4f0d130c58478468/lib/lib.dom.d.ts#L10069-L10097

Probably not worth a fight.